components/authz-service/config/project_update_stage.go

```diff
@@ -0,0 +1,141 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	event_ids "github.com/chef/automate/lib/event"
+)
+
+// ProjectUpdateStage - the config for project updating
+// The stage is all the data that needs to be remembered if the service is restarted.
+type ProjectUpdateStage struct {
+	State           string                       `toml:"state"`
+	ProjectUpdateID string                       `toml:"project_update_id"`
+	DomainServices  []ProjectUpdateDomainService `toml:"domain_services"`
+	FailureMessage  string                       `toml:"failure_message"`
+	Failed          bool                         `toml:"failed"`
+}
+
+func (stage ProjectUpdateStage) Copy() ProjectUpdateStage {
+	domainServices := make([]ProjectUpdateDomainService, len(stage.DomainServices))
+	for index, dService := range stage.DomainServices {
+		domainServices[index] = dService.copy()
+	}
+
+	return ProjectUpdateStage{
+		State:           stage.State,
+		ProjectUpdateID: stage.ProjectUpdateID,
+		DomainServices:  domainServices,
+		FailureMessage:  stage.FailureMessage,
+		Failed:          stage.Failed,
+	}
+}
+
+func (stage ProjectUpdateStage) StopRunning() ProjectUpdateStage {
+	stage.DomainServices = []ProjectUpdateDomainService{
+		ProjectUpdateDomainService{
+			Name:       event_ids.ComplianceInspecReportProducerID,
+			LastUpdate: time.Now(),
+		},
+		ProjectUpdateDomainService{
+			Name:       event_ids.InfraClientRunsProducerID,
+			LastUpdate: time.Now(),
+		},
+	}
+	stage.State = NotRunningState
+
+	return stage
+}
+
+func (stage ProjectUpdateStage) StartRunning(projectUpdateID string) ProjectUpdateStage {
+	stage.Failed = false
+	stage.FailureMessage = ""
+	stage.ProjectUpdateID = projectUpdateID
+	stage.DomainServices = []ProjectUpdateDomainService{
+		ProjectUpdateDomainService{
+			Name:       event_ids.ComplianceInspecReportProducerID,
+			LastUpdate: time.Now(),
+		},
+		ProjectUpdateDomainService{
+			Name:       event_ids.InfraClientRunsProducerID,
+			LastUpdate: time.Now(),
+		},
+	}
+	stage.State = RunningState
+
+	return stage
+}
+
+func (stage ProjectUpdateStage) AreDomainServicesComplete() bool {
+	for _, domainService := range stage.DomainServices {
+		if !domainService.Complete {
+			return false
+		}
+	}
+	return true
+}
+
+func (stage ProjectUpdateStage) HasFailedDomainService() bool {
+	failure := false
+	for _, domainService := range stage.DomainServices {
+		if domainService.Failed {
+			failure = true
```

Comment:
This could be `return true`, right?

Reply:
That would allow us to eliminate the var `failure`